crates/daphne/src/roles/mod.rs

```diff
@@ -91,45 +90,37 @@ async fn check_batch(
     Ok(())
 }
 
-async fn resolve_taskprov(
+async fn resolve_task_config(
     agg: &impl DapAggregator,
-    task_id: &TaskId,
     req: &DapRequestMeta,
-    global_config: &DapGlobalConfig,
-) -> Result<(), DapError> {
-    if agg.get_task_config_for(task_id).await?.is_some() {
-        // Task already configured, so nothing to do.
-        return Ok(());
-    }
-
-    let Some(vdaf_verify_key_init) = agg.taskprov_vdaf_verify_key_init() else {
-        warn!("Taskprov disabled due to missing VDAF verification key initializer.");
-        return Ok(());
-    };
-
-    let Some(collector_hpke_config) = agg.taskprov_collector_hpke_config() else {
-        warn!("Taskprov disabled due to missing Collector HPKE configuration.");
-        return Ok(());
-    };
-
-    let Some(task_config) = taskprov::resolve_advertised_task_config(
-        req,
-        vdaf_verify_key_init,
-        collector_hpke_config,
-        task_id,
-    )?
-    else {
-        // No task configuration advertised, so nothing to do.
-        return Ok(());
+) -> Result<DapTaskConfig, DapError> {
+    let task_config = if let Some(config) = agg.get_task_config_for(&req.task_id).await? {
+        config
+    } else {
+        let taskprov_config = agg.get_taskprov_config();
+        let (Some(taskprov_msg), Some(taskprov_config)) = (&req.taskprov, taskprov_config) else {
```

Comment:
Align with terminology in the spec
```suggestion
        let (Some(taskprov_advertisement), Some(taskprov_config)) = (&req.taskprov, taskprov_config) else {
```